pkg/eden/eden.go

```diff
@@ -436,19 +527,47 @@ func StartEVEQemu(qemuARCH, qemuOS, eveImageFile, qemuSMBIOSSerial string, eveTe
 	if qemuSMBIOSSerial != "" {
 		qemuOptions += fmt.Sprintf("-smbios type=1,serial=%s ", qemuSMBIOSSerial)
 	}
-	nets, err := utils.GetSubnetsNotUsed(2)
+	nets, err := utils.GetSubnetsNotUsed(1)
 	if err != nil {
 		return fmt.Errorf("StartEVEQemu: %s", err)
 	}
-	for ind, n := range nets {
-		qemuOptions += fmt.Sprintf("-netdev user,id=eth%d,net=%s,dhcpstart=%s", ind, n.Subnet, n.FirstAddress)
-		if ind == 0 {
-			for k, v := range qemuHostFwd {
-				qemuOptions += fmt.Sprintf(",hostfwd=tcp::%s-:%s", k, v)
-			}
+	offset := 0
+	firstNet := nets[0].Subnet
+	firstAddr := nets[0].FirstAddress
+	secondAddr := net.ParseIP("192.168.0.11")
```

Comment:
Existing logic created two different subnets and packets do not get routed between these subnets. Intention is to create a single subnet and keep both interfaces in that subnet. I did not understand the novel offsets part. Can you please elaborate?

Reply:
I mean, that before your changes we had nics on different subnets provided by `utils.GetSubnetsNotUsed` function. With your change we have one subnet (`firstNet`) which depends on environment (192.168.0.1/24 in my case) and another one is hardcoded (`net.ParseIP("192.168.0.11")`). Are we really expecting this? What is the point of putting a harcode in this line?
> I did not understand the novel offsets part. Can you please elaborate?

If I correctly understood, your add port forwarding on the second nic with predefined offset from ports in the config. In this case why not to add this logic into existing code
https://github.com/lf-edge/eden/blob/125938afbd50da1299d6ecdc97054b9f2e04353c/pkg/eden/eden.go#L447

>Intention is to create a single subnet and keep both interfaces in that subnet.

Could you elaborate why this is necessary?

Reply:
utils.GetSubnetsNotUsed creates subnet and that is 192.169.0.0/24 and it also allocates an IP i.e. 192.168.0.10 for the first NIC. The 192.168.0.11 that is see is not a subnet, but an IP used for the second NIC.
Both NICs have to be in the same subnet, because if they are in different subnets, qemu networking is not routing packets between them. Instead packets are going somewhere out into the internet. Probably that is because of the separate NAT instances created for these subnets. Also port forwarding has the case where port forwarded packets pass between two different interfaces in the same subnet. Basically apps connected to different network instances, but with uplink ports connected to the same network should be able to talk to each other using port forwarding.